### ISSUE TITLE
Investigate App Store, Windows Store, Launchpad.net (Debian/Ubuntu), Fedora

### ISSUE THREAD

**Issue by reynoldsalec** [kalabox/kalabox#1111]
- Get Certs
- See how the update process works. Can we send notifications?

**Comment by reynoldsalec** [comment]
## Mac App Store

tl;dr: I think the re-architecting around Apple's sandboxing requirements is probably the deal-killer here, at least for right now.

**Main Challenges:**

- App packaging
- App stability (for right now)
- Look + Feel requirements
- Sandboxing

**App Packaging**

I don't think this is a big hurdle:

- Add in https://github.com/johansatge/nwjs-macappstore-builder as a packaging option
- Make sure that the installer .pkg has the correct Identity key assigned (see https://github.com/nwjs/nw.js/wiki/MAS%3A-Uploading-the-binary for example)

**Look + Feel**

Further investigation of requirements would be helpful, but emulating a more standard Mac application interface is probably necessary for acceptance. Something like https://github.com/GoBijan/maverix would make this fairly easy.

**Sandboxing**
I think the Sandboxing requirement is the biggest concern; we'd probably have to rethink how we're handling a lot of the storage and information and other factors. See https://developer.apple.com/library/mac/documentation/Security/Conceptual/AppSandboxDesignGuide/AboutAppSandbox/AboutAppSandbox.html#//apple_ref/doc/uid/TP40011183-CH1-SW1 for an idea of how this would impact app architecture.

**Licensing:**

Currently, most of our licenses are...

- MIT
- ISC
- BSD or BSD-3-Clause
- Apache v3

I believe these are all compatible with the Mac App Store (the main problem we could have is from GNU).

**Comment by reynoldsalec** [comment]
## Windows Store

Seems to have far fewer requirements, although I found it more difficult to get more exact requirements from them.

I think we can start be testing the app with https://msdn.microsoft.com/en-us/windows/uwp/debug-test-perf/windows-app-certification-kit

**Comment by reynoldsalec** [comment]
## Ubuntu = Launchpad

https://developer.ubuntu.com/en/publish/other-forms-of-submitting-apps/ppa/

**Comment by pirog** [comment]
@reynoldsalec i'm hoping that a lot of the sandboxing requirements might be satisfied when we rebase on the new docker beta for mac/win
